test/unit/xml_state_test.py

```diff
@@ -199,6 +199,7 @@ def test_get_build_type_vmdisk_section(self):
     def test_get_build_type_vmnic_entries(self):
         assert self.state.get_build_type_vmnic_entries()[0].get_interface() \
             == ''
+        assert self.boot_state.get_build_type_vmnic_entries() == []
```

Comment:
Good, having 100% coverage is a very good measure but it certainly does not verify all the logic or execution *paths*. A while ago a looked if there was any tool to warn about not covered *elses* for unit tests, but did not find anything.